src/metrics.ts

```diff
@@ -0,0 +1,99 @@
+import { Registry, Counter, Gauge, collectDefaultMetrics } from 'prom-client'
+
+const METRIC_PREFIX = 'dex_price_estimator_'
+const BUY_AMOUNT_ESTIMATION_METRIC = METRIC_PREFIX + 'buy_amount_estimation'
+const ORDER_BOOK_METRIC = METRIC_PREFIX + 'order_book'
+const ORDER_BOOK_FETCHER_METRIC = METRIC_PREFIX + 'order_book_fetcher'
+
+export const register = new Registry()
+const baseMetricConfig = { registers: [register] }
+const marketsMetricConfig = { ...baseMetricConfig, labelNames: ['base', 'quote', 'hops'] }
+
+// Metrics for get-estimated-buy amount
+export const buyAmountEstimationMetrics = {
+  totalCount: new Counter({
+    name: BUY_AMOUNT_ESTIMATION_METRIC + '_requests_total',
+    help: 'Total requests of the buy amount estimation',
+    ...baseMetricConfig,
+  }),
+
+  count: new Counter({
+    name: BUY_AMOUNT_ESTIMATION_METRIC + '_requests',
+    help: 'Requests of the buy amount estimation by market and hops',
+    ...marketsMetricConfig,
+  }),
+
+  errorsCount: new Counter({
+    name: BUY_AMOUNT_ESTIMATION_METRIC + '_error_total',
+    help: 'Number of failed requests of the buy amount estimation',
+    ...marketsMetricConfig,
+  }),
+
+  durationsTotals: new Gauge({
+    name: BUY_AMOUNT_ESTIMATION_METRIC + '_duration_seconds_total',
+    help: 'Request duration of the buy amount estimation in seconds',
+    ...baseMetricConfig,
+  }),
+
+  durations: new Gauge({
+    name: BUY_AMOUNT_ESTIMATION_METRIC + '_duration_seconds',
+    help: 'Request duration of the buy amount estimation in seconds by market and hops',
+    ...marketsMetricConfig,
+  }),
+}
+
+export const getOrderBookMetrics = {
```

Comment:
nit: I'm surprised that this is called get even though it's not a function.

Reply:
agree, I though several times about it, cause it was weird for me too. It's get, like http "GET order book metrics", but I think it can make things confusing, so renaming